lib/Ceilometer/Fold.hs

```diff
@@ -234,4 +292,11 @@ eEvent (start, end) = M.foldlWithKey (\a k v -> a + (fromIntegral k * v)) 0 . go
         go a@_                          = snd a
         s x = max start (x ^. time)
 
+eIPEvent (start, end) = M.foldlWithKey (\a _ v -> a + v) 0 . go . unwrapAcc
+  where -- deal with the dangling last event
+        go (Just x, acc)
+          | d <- end - s x, d > 0 = insertEvent x d acc
+        go a@_                          = snd a
+        s x = max start (x ^. time)
+
```

Comment:
this is finicky logic and I'd like to not see it duplicated.

instead, can you change the end event step (eEvent) to take a fold function (acc -> k -> v -> acc) and use eEvent for everything?